### PR TITLE
Set errno if fail to open file from Android asset manager

### PR DESCRIPTION
**Pull request by jonastsai**
b/377488323

### CODE REVIEW

File: starboard/android/shared/file_internal.cc [hunk 1]
```diff
@@ -112,10 +112,15 @@ bool IsAndroidAssetPath(const char* path) {
 
 AAsset* OpenAndroidAsset(const char* path) {
   if (!IsAndroidAssetPath(path) || g_asset_manager == NULL) {
+    errno = ENOENT;
```
[the remark this answers is not in the document]
Comment (jonastsai) [reply starboard/android/shared/file_internal.cc:116]: okay, I add a debug warning message here

File: starboard/android/shared/file_internal.cc [hunk 2]
```diff
@@ -112,10 +112,15 @@ bool IsAndroidAssetPath(const char* path) {
...
     return NULL;
   }
   const char* asset_path = path + strlen(g_app_assets_dir) + 1;
-  return AAssetManager_open(g_asset_manager, asset_path, AASSET_MODE_RANDOM);
+  AAsset* result = AAssetManager_open(g_asset_manager, asset_path, AASSET_MODE_RANDOM);
```
[the remark this answers is not in the document]
Comment (jonastsai) [reply starboard/android/shared/file_internal.cc:120]: no. I cannot find the document about it. the errno seems more likely for POSIX API. Android API doesn't seems to compliant with it.

**Comment by kaidokert** [comment]
I'll defer to @hlwarriner and @yjzhang111 for the review here. Thanks for tracking the fix !

File: starboard/android/shared/file_internal.cc [hunk 3]
```diff
@@ -112,10 +112,17 @@ bool IsAndroidAssetPath(const char* path) {
 
 AAsset* OpenAndroidAsset(const char* path) {
   if (!IsAndroidAssetPath(path) || g_asset_manager == NULL) {
+    SB_DLOG(WARNING) << "Unable to open from Android Asset Manager: " << path;
```
Comment (kaidokert) [inline starboard/android/shared/file_internal.cc:115]: Change those to `SB_LOG` please so that they are always present, outside of debug builds as well.

Comment (jonastsai) [reply starboard/android/shared/file_internal.cc]: Sure. it is updated to use `SB_LOG`

**Pull request merged**